packages/protocol/contracts/L1/verifiers/SgxVerifier2.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.20;
+
+import { ECDSAUpgradeable } from "@ozu/utils/cryptography/ECDSAUpgradeable.sol";
+
+import { EssentialContract } from "../../common/EssentialContract.sol";
+import { Proxied } from "../../common/Proxied.sol";
+import { LibBytesUtils } from "../../thirdparty/LibBytesUtils.sol";
+
+import { TaikoData } from "../TaikoData.sol";
+
+import { IVerifier } from "./IVerifier.sol";
+
+/// @title SgxVerifier2
+contract SgxVerifier2 is EssentialContract, IVerifier {
```

Comment:
This contract lacking the following functionality:
- registerInstancesByAnotherSGXInstance (Please see `addToRegistryBySgxInstance` in `SgxVerifier.sol`)


I guess the one below is somewhat the one, but it lacks the functionality to add, other trusted party's address.     
```
// TODO(dani): who will call this function?
    function replaceInstance(
        address newInstance,
        bytes memory signature
    )
        external
    {
        address oldInstance = keccak256(
            abi.encode("REGISTER_SGX_INSTANCE", newInstance)
        ).recover(signature);

        _replaceInstance(oldInstance, newInstance);
    }
```

Reply:
Beside that LGTM.

Reply:
Dani, why the SGX instance wants to add other third party addresses? Should we just use the owner's registerInstance() function?

Reply:
Hey Daniel,
So implementing Brecht’s proposal , see as:

“Still an interesting alternative approach without any onchain attestion verification: just use a set of initial set of "bootstrap" SGX nodes when we launch the rollup that have published their attestation somewhere and so everyone can check it offchain. These bootstrap SGX nodes can then verify attestations from other SGX machines that want to generate proofs (and these themselves are also allowed to verify attestations from others). So this way onchain we simply just have to check a simple signature for SGX machines that want to register themselves so their SGX proofs can be accepted.”

Also see the reference here in the first branch:

https://github.com/taikoxyz/taiko-mono/pull/14851#discussion_r1340351054

Reply:
I still don't know the whole SGX solution well, so lets skip this question and leave it to you, Patryk, and Brecht. 

The new implementation is simply a more concise implementation of SGXVerifier.sol